File: src/main/java/br/gov/servicos/editor/servicos/Etapa.java
package br.gov.servicos.editor.servicos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Wither;

import java.util.List;

@Data
@Wither
@NoArgsConstructor
@AllArgsConstructor
public class Etapa {

    String titulo;
    String descricao;
    List<Custo> custos;
    Boolean custoTemExcecoes;
    List<String> documentos;
    List<CanalDePrestacao> canaisDePrestacao;

}

File: src/main/resources/static/editar/assets/componentes/etapa/etapa.js
var etapa = etapa || {};

var Etapa = {

  controller: function(args) {
    this.etapa = args.etapa;
    this.gratuidade = args.gratuidade;
  },

  view: function(ctrl) {
    return m('.etapa#' + ctrl.etapa.id, [
      m.component(etapa.Titulo, { titulo: ctrl.etapa.titulo }),
      m.component(etapa.Descricao, { descricao: ctrl.etapa.descricao }),
      m.component(etapa.Documentos , { documentos: ctrl.etapa.documentos }),
      ctrl.gratuidade() ? null : m.component(etapa.Custos, { custos: ctrl.etapa.custos }),
      m.component(etapa.CanaisDePrestacao, { canaisDePrestacao: ctrl.etapa.canaisDePrestacao }),
    ])
  }
};
